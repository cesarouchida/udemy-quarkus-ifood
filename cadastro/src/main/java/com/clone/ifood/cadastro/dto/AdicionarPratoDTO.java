package com.clone.ifood.cadastro.dto;

import com.clone.ifood.cadastro.Restaurante;
import com.clone.ifood.cadastro.infra.DTO;
import com.clone.ifood.cadastro.infra.ValidDTO;

import javax.validation.ConstraintValidatorContext;
import java.math.BigDecimal;

@ValidDTO
public class AdicionarPratoDTO implements DTO {
    public String nome;
    public String descricao;
    public BigDecimal preco;

    @Override
    public boolean isValid(ConstraintValidatorContext constraintsConstraintValidatorContext) {
        return DTO.super.isValid(constraintsConstraintValidatorContext);
    }
}
