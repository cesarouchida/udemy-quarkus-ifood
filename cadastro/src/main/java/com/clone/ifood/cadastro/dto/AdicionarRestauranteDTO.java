package com.clone.ifood.cadastro.dto;

import com.clone.ifood.cadastro.Restaurante;
import com.clone.ifood.cadastro.infra.DTO;
import com.clone.ifood.cadastro.infra.ValidDTO;
import org.hibernate.validator.constraints.br.CNPJ;

import javax.validation.ConstraintValidatorContext;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

@ValidDTO
public class AdicionarRestauranteDTO implements DTO {
    @NotEmpty
    @NotNull
    public String proprietario;

    @CNPJ
    @NotNull
    public String cnpj;

    @Size(min=3, max=30)
    public String nomeFantasia;
    public LocalizacaoDTO localizacao;

    @Override
    public boolean isValid(ConstraintValidatorContext constraintsConstraintValidatorContext) {
        constraintsConstraintValidatorContext.disableDefaultConstraintViolation();
        if(Restaurante.find("cnpj", cnpj).count() > 0) {
            constraintsConstraintValidatorContext
                    .buildConstraintViolationWithTemplate("CNPJ jรก cadastrado")
                    .addPropertyNode("cnpj")
                    .addConstraintViolation();
            return false;
        }
        return true;
    }
}
