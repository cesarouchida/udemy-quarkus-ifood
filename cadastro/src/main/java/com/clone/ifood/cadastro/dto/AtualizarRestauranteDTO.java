package com.clone.ifood.cadastro.dto;

public class AtualizarRestauranteDTO {
    public String nomeFantasia;
}
