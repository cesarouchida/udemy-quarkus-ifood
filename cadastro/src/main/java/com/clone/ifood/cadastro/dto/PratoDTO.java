package com.clone.ifood.cadastro.dto;

import org.mapstruct.Mapping;

import java.math.BigDecimal;

public class PratoDTO {
    public Long id;
    public String nome;
    public String descricao;
    public RestauranteDTO restaurante;
    public BigDecimal preco;
}
