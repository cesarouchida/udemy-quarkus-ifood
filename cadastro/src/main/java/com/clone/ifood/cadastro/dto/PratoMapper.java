package com.clone.ifood.cadastro.dto;

import com.clone.ifood.cadastro.Prato;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.MappingTarget;
import org.mapstruct.Mappings;

@Mapper(componentModel = "cdi")
public interface PratoMapper {

    Prato toPrato(AdicionarPratoDTO dto);

    void toPrato(AtualizarPratoDTO dto, @MappingTarget Prato prato);

    PratoDTO toPratoDTO(Prato prato);
}
