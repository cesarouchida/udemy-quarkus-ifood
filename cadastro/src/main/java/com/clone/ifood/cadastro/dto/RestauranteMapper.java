package com.clone.ifood.cadastro.dto;

import com.clone.ifood.cadastro.Restaurante;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.MappingTarget;

@Mapper(componentModel = "cdi")
public interface RestauranteMapper {

    @Mapping(target = "nome", source="nomeFantasia")
    Restaurante toRestaurante(AdicionarRestauranteDTO dto);

    @Mapping(target = "nome", source="nomeFantasia")
    void toRestaurante(AtualizarRestauranteDTO dto, @MappingTarget Restaurante restaurante);

    @Mapping(target = "nomeFantasia", source="nome")
    @Mapping(target = "dataCriacao", dateFormat = "dd/MM/yyyy HH:mm:ss")
    RestauranteDTO toRestauranteDTO(Restaurante restaurante);
}
