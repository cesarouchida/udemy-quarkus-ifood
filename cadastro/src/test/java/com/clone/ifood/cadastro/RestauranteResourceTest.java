package com.clone.ifood.cadastro;

import com.clone.ifood.cadastro.dto.AdicionarRestauranteDTO;
import com.clone.ifood.cadastro.dto.AtualizarRestauranteDTO;
import com.clone.ifood.cadastro.util.TokenUtils;
import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import io.restassured.RestAssured;
import io.restassured.http.ContentType;
import io.restassured.http.Header;
import io.restassured.specification.RequestSpecification;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static javax.ws.rs.core.Response.Status;
import static org.junit.jupiter.api.Assertions.assertEquals;

@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(CadastroTestLifecycleManager.class)
class RestauranteResourceTest {

    private String token;

    // Gerando token para os Testes
    @BeforeEach
    public void gereToken() throws Exception {
        token = TokenUtils.generateTokenString("/JWTProprietarioClaims.json", null );
    }

    // Uso padrão para given Informando o contentType
    // Implementação do token para os testes
    private RequestSpecification given(){
        return RestAssured.given()
                .contentType(ContentType.JSON)
                .header(new Header("Authorization", "Bearer " + token));
    }

    @Test
    @DataSet("restaurantes-cenario-1.yml")
    void buscarRestaurantes(){
        var resultado = given()
            .when().get("/restaurantes")
            .then()
            .statusCode(Status.OK.getStatusCode())
            .extract().asString();
        Approvals.verifyJson(resultado);
    }

    @Test
    void inseriRestaurante(){
        AdicionarRestauranteDTO dto = new AdicionarRestauranteDTO();
        dto.cnpj = "43.843.541/0001-94";
        dto.nomeFantasia = "Potato";
        dto.proprietario = "Joao";

        given()
            .body(dto)
            .when().post("/restaurantes")
            .then()
            .statusCode(Status.CREATED.getStatusCode());

        Restaurante findById = Restaurante.findById(1L);
//        System.out.println(Restaurante.findAll().stream().count());


        assertEquals(dto.nomeFantasia, findById.nome);
    }

    @Test
    @DataSet("restaurantes-cenario-1.yml")
    void atualizaRestaurante(){
        AtualizarRestauranteDTO dto = new AtualizarRestauranteDTO();
        dto.nomeFantasia = "Potato";
        Long idRestaurante = 123L;
        given()
                .with().pathParam("id", idRestaurante)
                .body(dto)
                .when().put("/restaurantes/{id}")
                .then()
                .statusCode(Status.NO_CONTENT.getStatusCode());

        Restaurante findById = Restaurante.findById(idRestaurante);

        assertEquals(dto.nomeFantasia, findById.nome);
    }

    @Test
    @DataSet("restaurantes-cenario-1.yml")
    void deleteRestaurante(){
        Long idRestaurante = 123L;

        given()
                .with().pathParam("id", idRestaurante)
                .when().delete("/restaurantes/{id}")
                .then()
                .statusCode(Status.NO_CONTENT.getStatusCode());

        assertEquals(Restaurante.count(), 1);
    }


}


