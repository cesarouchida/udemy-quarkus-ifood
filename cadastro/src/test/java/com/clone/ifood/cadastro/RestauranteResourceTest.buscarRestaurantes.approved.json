[
  {
    "id": 123,
    "nomeFantasia": "Novo Restaurante",
    "proprietario": "ID do keycloak"
  },
  {
    "id": 234,
    "nomeFantasia": "Novo Restaurante 2",
    "proprietario": "ID do keycloak 2"
  }
]